cert/aziot-certd/src/lib.rs

```diff
@@ -106,10 +115,12 @@ struct Api {
     cert_issuance: CertIssuance,
     preloaded_certs: BTreeMap<String, PreloadedCert>,
     principals: BTreeMap<libc::uid_t, Vec<wildmatch::WildMatch>>,
+    renewal_engine: Arc<Mutex<cert_renewal::RenewalEngine<renewal::EstIdRenewal>>>,
 
     key_client: Arc<aziot_key_client_async::Client>,
-    key_engine: FunctionalEngine,
-    proxy_uri: Option<hyper::Uri>,
+    key_engine: Arc<Mutex<FunctionalEngine>>,
```

Comment:
You should put this inside regular `std::sync::Mutex` instead of the async one, since it doesn't have any async API of its own so the mutex can be acquired and released synchronously.

Reply:
Fixed